code/game/machinery/computer/station_alert.dm

```diff
@@ -3,16 +3,50 @@
 	desc = "Used to access the station's automated alert system."
 	icon_screen = "alert:0"
 	icon_keyboard = "atmos_key"
-	circuit = /obj/item/circuitboard/computer/stationalert
+	circuit = /obj/item/circuitboard/computer/station_alert
 	light_color = LIGHT_COLOR_CYAN
 	/// Station alert datum for showing alerts UI
 	var/datum/station_alert/alert_control
+	/// Determines if the alerts track via z-level, or only show all station areas
+	var/station_only
+
+/obj/machinery/computer/station_alert/examine(mob/user)
+	. = ..()
+	. += span_info("The console is set to [station_only ? "track all station and mining alarms" : "track alarms on the same z-level"].")
 
 /obj/machinery/computer/station_alert/Initialize(mapload)
-	alert_control = new(src, list(ALARM_ATMOS, ALARM_FIRE, ALARM_POWER), list(z), title = name)
+	// Checks the internal circuit to determine if it's been altered by players to (or spawned in) station_only mode
+	var/obj/item/circuitboard/computer/station_alert/my_circuit = circuit
+	station_only = my_circuit.station_only
+
+	// If the console is station_only, set up the alert_control to only listen to station areas
+	// This only functions for mapped in ones. Ones built in-round have to redo this in on_construction().
+	if(station_only)
+		name = "station alert console"
+		var/list/alert_areas
+		alert_areas = (GLOB.the_station_areas + typesof(/area/mine))
+		alert_control = new(src, list(ALARM_ATMOS, ALARM_FIRE, ALARM_POWER), listener_areas = alert_areas, title = name)
+	else // If not station_only, just check the z-level
+		name = "local alert console"
+		alert_control = new(src, list(ALARM_ATMOS, ALARM_FIRE, ALARM_POWER), list(z), title = name)
 	RegisterSignals(alert_control.listener, list(COMSIG_ALARM_LISTENER_TRIGGERED, COMSIG_ALARM_LISTENER_CLEARED), PROC_REF(update_alarm_display))
 	return ..()
 
+/obj/machinery/computer/station_alert/on_construction(mob/user)
+	. = ..()
+	var/obj/item/circuitboard/computer/station_alert/my_circuit = circuit
+	station_only = my_circuit.station_only
+	// The circuitboard isn't "added" to the machine until this proc... this code resets and re-does
+	// the code done in Initialize, so that station crew can re-build these during a round.
```

Comment:
was this your worry?

Reply:
Yeah, this is it. It's kind of brute-forceing it by unregistering the listener just to get the new one set up.
I couldn't get it to set up the listener any later than initialize or icons and other procs got upset there was none, but setting it up any earlier would happen before the circuitboard was in so it never read the station variable.